packages/video_player/video_player/lib/src/web_vtt.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:html/dom.dart';
+
+import 'closed_caption_file.dart';
+import 'package:html/parser.dart' as html_parser;
+
+/// Represents a [ClosedCaptionFile], parsed from the WebVtt file format.
+/// See: https://en.wikipedia.org/wiki/WebVtt
+class WebVttCaptionFile extends ClosedCaptionFile {
+  /// Parses a string into a [ClosedCaptionFile], assuming [fileContents] is in
+  /// the WebVtt file format.
+  /// * See: https://en.wikipedia.org/wiki/WebVtt
+  WebVttCaptionFile(this.fileContents)
+      : _captions = _parseCaptionsFromWebVttString(fileContents);
+
+  /// The entire body of the Vtt file.
+  final String fileContents;
+
+  @override
+  List<Caption> get captions => _captions;
+
+  final List<Caption> _captions;
+}
+
+List<Caption> _parseCaptionsFromWebVttString(String file) {
+  final List<Caption> captions = <Caption>[];
+
+  /// Ignore metadata
+  List<String> metadata = ['HEADER', 'NOTE', 'REGION', 'WEBVTT'];
+
+  int captionNumber = 1;
+  for (List<String> captionLines in _readWebVttFile(file)) {
+    /// CaptionLines represent a complete caption
```

Comment:
Thanks for adding this comment! We usually use // for regular comments. Do you mind fix it too>